MGDAPI-3855 - Implement persistent volume for alert manager

valerymo opened the pull request:
# Issue link
Implement persistent volume for alert manager
https://issues.redhat.com/browse/MGDAPI-3855
This PR could be merged as Observability PR (that part of this Jira): https://github.com/redhat-developer/observability-operator/pull/114 is already merged and New OO release available: v3.0.11.

# What
- Implement persistent volume for alert manager
- This PR includes change of OO version: from v3.0.8 to v3.0.11.

# Verification steps
1. Check that Alertmanager PVC created in redhat-rhoam-observability namespace. For example:
$ oc get pvc -n redhat-rhoam-observability
NAME STATUS VOLUME CAPACITY ACCESS MODES STORAGECLASS AGE
alertmanager-alertmanager-db-alertmanager-alertmanager-0 Bound pvc-60903afd-b8b0-43c1-8fc4-1adc4062fe01 1Mi RWO gp2 3h2m
prometheus-prometheus-db-prometheus-prometheus-0 Bound pvc-4ae289f2-5fa8-42f1-9864-19aa805e0760 50Gi RWO gp2 3h2m

2. Enter Alertmanager UI, Silent/Snooze any alert (for example alertname="DeadMansSwitch"), kill alertmanager pod , wait for recreation of the pod, enter again Alertmanager UI, and see that alert still snoozed

3. Notes: - The PR was verified on local installation and install from bundle  - https://quay.io/repository/vmogilev_rhmi/managed-api-service-index?tab=tags&tag=1.23.0

KevFan commented on pkg/products/observability/reconciler.go:
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
 	prometheus "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	observability "github.com/redhat-developer/observability-operator/v3/api/v1"
+	//observability1 "github.com/valerymo/observability-operator/v3/api/v1"
```
I think we can remove this import as it looks to be duplicated :thinking:

valerymo replied:
Done. Thank you

makslion commented:
Should this PR be tested/merged after [this](https://github.com/redhat-developer/observability-operator/pull/114) one gets through? :thinking:

valerymo commented:
Subject updated. Yes, this PR should be merged after OO PR https://github.com/redhat-developer/observability-operator/pull/114

valerymo commented:
@KevFan , @makslion , sorry , can't find option in my browser for reply on your comments.
1- Done, unused/commented import - removed
2 - Subject updated. Yes, this PR should be merged after OO PR https://github.com/redhat-developer/observability-operator/pull/114
Thanks very much for your review and comments

valerymo commented:
waiting for new OO release, that include OO PR https://github.com/redhat-developer/observability-operator/pull/114

valerymo commented:
@makslion , @KevFan ,  PR updated - new OO release v3.0.11, that includes second PR (in OO) of current task - https://github.com/redhat-developer/observability-operator/pull/114.     Verification done on cluster `valerym01` (cluster is available till 2022-06-16 , 16:00). Could you please review + lgtm

KevFan commented:
Verified, looks good :+1: 

One thing is I think we should also include the OO manifest from https://github.com/redhat-developer/observability-operator/tree/v3.0.11/bundle/manifests to https://github.com/integr8ly/integreatly-operator/tree/master/manifests/integreatly-observability-operator :thinking:

valerymo commented:
@KevFan , @makslion  thank you very much for comments!  I forgot (didn't know : ) about manifests ... So copied now, but found that in observability-operator.clusterserviceversion.yaml appears version 3.0.10 instead of 3.0.11, so checking now with Vincent. Thanks again for comments

valerymo commented:
@KevFan ,  Vincent updated clusterserviceversion manifest in v3.0.11, but anyway I needed to change replaced release from 3.0.10 to 3.0.8. Hope it's ok (?). Thanks again for review and comments.

KevFan commented:
@valerymo Yeah I think that's fine. I think the main value for the manifests are for when generating the related images when cutting a release :+1: 

This looks good to me anyway :+1: 

/lgtm

KevFan commented:
/lgtm

KevFan commented:
/test rhoam-e2e
/lgtm

KevFan commented:
/test rhoam-e2e

KevFan commented:
/lgtm cancel 

@valerymo Looks like there's a failing e2e test if you can address that also. Looks to be the addition of the new PVC :thinking:  

```
    A06 - Verify PVC [It]
    /go/src/github.com/integr8ly/integreatly-operator/test/e2e/integretly_test.go:69
    Persistant Volume Claim: prometheus-prometheus-db-prometheus-prometheus-0 not found

```

valerymo commented:
@KevFan , thanks again for issue found. I updated test/common/pvc_validate.go.  Name of PVC was changed in new OO version, so i updated it now; and also added new PVC to test. These is how it now appears in my cluster `valerym0`:
[integreatly-operator (valery_MGDAPI-3855)]$ oc get pvc -n redhat-rhoam-observability
NAME                                                       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
alertmanager-alertmanager-db-alertmanager-alertmanager-0   Bound    pvc-c739bdd3-2862-411d-93d9-fe8b35b67e11   1Gi        RWO            gp2            53m
managed-services-prometheus-prometheus-0                   Bound    pvc-73072d50-6885-416b-bc7b-f934816a7a79   50Gi       RWO            gp2            53m

Locally A06 was tested. Here tests are running and will check it

KevFan commented:
/test rhoam-e2e

makslion commented:
Looks fine 

/lgtm

valerymo commented:
/test rhoam-e2e

valerymo commented:
/test rhoam-e2e

makslion commented:
/retest

valerymo commented:
/test rhoam-e2e

makslion commented:
tests seems to be fixed 

/lgtm

MStokluska commented on products/installation.yaml:
```diff
@@ -106,7 +106,7 @@ products:
     package: observability-operator
     installFrom: "index"
-    index: "quay.io/rhoas/observability-operator-index:v3.0.10"
+    index: "quay.io/rhoas/observability-operator-index:v3.0.11"
```
@valerymo does the installation-cpaas also require a bump?

valerymo replied:
@MStokluska , Thank you very much for this comment. installation-cpaas - updated.

valerymo commented:
/test rhoam-e2e
